src/Encryption.php

```diff
@@ -328,8 +325,12 @@ private static function calculateAgreementKey(JWK $private_key, JWK $public_key)
     {
         if (function_exists('openssl_pkey_derive')) {
             try {
-                $publicPem = ECKey::convertPublicKeyToPEM($public_key);
-                $privatePem = ECKey::convertPrivateKeyToPEM($private_key);
+                $publicPem = openssl_pkey_get_public(ECKey::convertPublicKeyToPEM($public_key));
```

Comment:
Is this new? Why is there openssl_pkey_get_public and openssl_pkey_get_private now?

Reply:
I don't fully remember whether I had some issues relating to this. However, I think I decided to use this as an extra safety net before passing the keys to openssl_pkey_get_private(). See also https://www.php.net/manual/en/openssl.certparams.php.
In this case I don't have any preference and can remove it as well.

Reply:
Removed that part as it's a bit out of scope and I think phpstan also does not complain with or without it.